src/cmds.c

```diff
@@ -853,20 +853,21 @@ static void cmd_pls_bot(struct userrec *u, int idx, char *par)
   }
 
 #ifndef TLS
-  if ((port && *port == '+') || (relay && relay[0] == '+')) {
-    dprintf(idx, "Ports prefixed with '+' are not enabled "
+  if ((port && (*port == '+' || *port == '-')) ||
+      (relay && (*relay == '+' || *relay == '-'))) {
+    dprintf(idx, "Ports prefixed with '+' or '-' are not enabled "
       "(this Eggdrop was compiled without TLS support).\n");
     return;
   }
 #endif
   if (port) {
-    if (!check_int_range(port, 0, 65536)) {
+    if (!check_int_range(port, 0, 65536) && !check_int_range(port, -65536, 0)) {
       dprintf(idx, "Ports must be integers between 1 and 65535.\n");
       return;
     }
   }
   if (relay) {
-    if (!check_int_range(relay, 0, 65536)) {
+    if (!check_int_range(relay, 0, 65536) && !check_int_range(relay, -65536, 0)) {
```

Comment:
maybe just check_int_range(port, -65536, 65536)?

Reply:
no, that would include 0 which is invalid